### PR TITLE
#163426526 article CRUD operations

### PR DESCRIPTION
**Pull request by ozone4real**
#### What does this PR do?
It creates API endpoints for article CRUD operations.

#### Description of Task to be completed?
Have the following endpoints working:
- Create an article: `POST /api/v1/articles`
- Get all articles: `GET /api/v1/articles`
- Get all articles written by a specific user: `GET /api/v1/articles/user/:username`
- Get a single article: `GET /api/v1/articles/:slug`
- Update an article: `PUT /api/v1/articles/:slug/update`
- Delete an article: `DELETE api/v1/articles/:slug/delete`

#### How should this be manually tested?
Clone this repo, cd into it and run npm install.
 **With test suite:**
     - Run `npm test`

 **Using postman:**
     - Run `npm start`,
     - With postman, send requests to `localhost:3000/<api url>` with the relevant HTTP methods and request payloads.

#### What are the relevant pivotal tracker stories?
[#163426526](https://www.pivotaltracker.com/story/show/163426526)

### CODE REVIEW

**Comment by omenkish** [comment]
why did the coverage reduce with over 3% @ozone4real ?

**Comment by ozone4real** [comment]
> why did the coverage reduce with over 3% @ozone4real ?

@omenkish the test coverage for my branch is 98.12%. I'm not sure how that -3% came about

**Review by divinediscipline — DISMISSED** [review]
Good work.

**Review by EfeAgare — DISMISSED** [review]

**Review by biolabalo — DISMISSED** [review]

**Review by omenkish — DISMISSED** [review]

File: server/controllers/articleController.js [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+          {
+            model: Users,
+            as: 'author',
+            attributes: ['username', 'bio', 'imageUrl']
+          },
+          { model: Comments },
+          { model: Reactions }
+        ]
+      });
+      if (!articles[0]) return errorResponse(res, 404, 'No articles found');
```
Comment (p-funky) [inline server/controllers/articleController.js]: What is the value of `articles` if the findAll query does not find anything? Is it `undefined` or an empty array?

Comment (ozone4real) [reply server/controllers/articleController.js]: An empty array

Comment (p-funky) [reply server/controllers/articleController.js]: I'll prefer checking with `!articles.length` then. [].length = 0 which is falsy in javascript. All other numbers are truthy.

Comment (ozone4real) [reply server/controllers/articleController.js]: Okay. I thought (articles.length === 0) and (!articles[0]) does the same thing. So I should change to (articles.length ===0) ?

Comment (p-funky) [reply server/controllers/articleController.js]: Yes they do the same thing but I am saying just change it to `!articles.length` actually.

Comment (ozone4real) [reply server/controllers/articleController.js]: Okay. On it. Thanks

**Review by omenkish — APPROVED** [review]

**Review by p-funky — APPROVED** [review]

**Pull request merged**